nbbuild/antsrc/org/netbeans/nbbuild/reporting/RatHtmlReportTask.java

```diff
@@ -0,0 +1,295 @@
+            try {
+                fileWriter = new FileWriter(targetFile);
+                bw = new BufferedWriter(fileWriter);
+                bw.write("<!DOCTYPE html>");
+                bw.write("<html>");
+                bw.write("<head>");
+                writeStyle(bw);
+                bw.write("</head>");
+                bw.write("<body>");
+                bw.write("<div id=\"content\">");
+                writeAreas(bw, clusterList, modulebycluster, moduleRATInfo);
+                bw.write("</div>");
+                bw.write("<nav id=\"main-menu\">");
+                writeMenu(bw, clusterList, modulebycluster, moduleRATInfo);
+                bw.write("</nav>");
+                bw.write("</body>");
+                bw.write("</html>");
+
+            } catch (IOException ex) {
+                Logger.getLogger(RatHtmlReportTask.class.getName()).log(Level.SEVERE, null, ex);
+            } finally {
+                try {
+                    bw.close();
+                    fileWriter.close();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+
+            }
+        }
+    }
+
+    private static void writeMenu(BufferedWriter bw, Set<String> clusterDB, Map<String, Set<String>> modulebycluster, Map<String, ModuleInfo> moduleRATInfo) throws IOException {
+        bw.write("<ul>");
+        for (String clusterName : clusterDB) {
+            int appscanned = 0;
+            int unappscanned = 0;
+            for (String mm : modulebycluster.get(clusterName)) {
+                ModuleInfo get = moduleRATInfo.get(mm);
+                appscanned += get.getApprouved().size();
+                unappscanned += get.getUnapprouved().size();
+            }
+            if ((unappscanned + appscanned) > 0) {
+                bw.write("<li>");
+            } else {
+                bw.write("<li class=\"notpresent\">");
+            }
+            bw.write(clusterName);
+            bw.write(" (A:" + appscanned + ", U:" + unappscanned + ")</li>");
+        }
+        bw.write("</ul>");
+    }
+
+    private void writeAreas(BufferedWriter bw, Set<String> clusterDB, Map<String, Set<String>> modulebycluster, Map<String, ModuleInfo> unbymodule) throws IOException {
+        for (String clusterName : clusterDB) {
+            bw.write("<div class=\"area\" id=\"" + clusterName + "\">");
+            bw.write("<h2>Cluster::" + clusterName + "</h2>");
+            for (String mm : modulebycluster.get(clusterName)) {
+                bw.write("<div class=\"module\" id=\"" + mm + "\">");
+                bw.write("<h3>Module::" + mm + "</h3>");
+                bw.write("<ul>");
+                ModuleInfo get = unbymodule.get(mm);
+                for (String mmu : get.getUnapprouved()) {
+                    bw.write("<li class=\"unapprouved\">");
+                    bw.write("U:<a href=\"https://github.com/apache/incubator-netbeans/tree/master/" + mmu + "\">" + mmu + "</a>");
```

Comment:
Hardcoding repository URL is probably not good idea.

Reply:
nice addition thanks. will look at it